api/v1/storagecluster_types.go

```diff
@@ -162,9 +162,11 @@ type ExternalStorageClusterSpec struct {
 	// StorageProviderKind Identify the type of storage provider cluster this consumer cluster is going to connect to.
 	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
 
-	// ConnectionString An encoded string holding connection and identity information
-	// that is needed in order to establish connection with the storage providing cluster.
-	ConnectionString string `json:"connectionString,omitempty"`
```

Comment:
The code that removes usage of  `ConnectionString` should  also have been in the same commit. Right now its removed in the third commit.

Reply:
Actually, we like to keep API and controller changes in the different commits.

Reply:
This is a different scenario.  An API field is removed that is being used in the controller.  This commit on its own won't build. IMO, `ConnectionString` and all its references should have been removed in a single PR. 
Approving the PR.